### PR TITLE
Make TaskContainer.whenObjectRemoved() an error

### PR DESCRIPTION
**Pull request by ghale**

### CODE REVIEW

File: subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java [hunk 1]
```diff
@@ -85,18 +84,6 @@ public String create() {
...
                         }
                     });
-                    DeprecationLogger.whileDisabled(new Runnable() {
-                        @Override
-                        public void run() {
-                            container.whenObjectRemoved(new Action<I>() {
-                                @Override
-                                public void execute(I item) {
-                                    String name = namer.determineName(item);
-                                    containerNode.removeLink(name);
```
Comment (big-guy) [inline subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java]: Is whenObjectRemoved still called when a task is replaced?

Comment (ghale) [reply subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java]: Why yes, yes it is...fixed.

File: subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java [hunk 2]
```diff
@@ -88,7 +90,7 @@ public String create() {
                     DeprecationLogger.whileDisabled(new Runnable() {
```
Comment (big-guy) [inline subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java]: I think this isn't needed now

Comment (ghale) [reply subprojects/core/src/main/java/org/gradle/model/collection/internal/BridgedCollections.java]: Right on.  Removed.

**Pull request merged**